vault/router.go

```diff
@@ -62,6 +62,13 @@ func (r *Router) Mount(backend logical.Backend, prefix string, mountEntry *Mount
 	if existing, _, ok := r.root.LongestPrefix(prefix); ok && existing != "" {
 		return fmt.Errorf("cannot mount under existing mount '%s'", existing)
 	}
+	// If this is a secret backend, check to see if the prefix conflicts
+	// with an existing mountpoint
+	if prefix != "" {
+		if match := r.matchingPrefixInternal(prefix); match != "" {
```

Comment:
I think my preference here would be to allow it in the router but not the mount code. So existing setups will work, but it won't be allowed for new mounts.